airflow/utils/analysis.py

```diff
@@ -0,0 +1,38 @@
+# -*- coding: utf-8 -*-
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+def gen_airflow_analysis_metadata_dict(context):
```

Comment:
I think it can be made even more generic. This class takes a context and returns a dict of key airflow vars so how about context_to_airflow_vars (and same for the filename). Ideally this would be a method in a context class of some sort but it's probably OK to just leave it here for now.

The Dr. Elephant stuff can be removed from the comments.